### PR TITLE
Added instructions for bind

### PR DESCRIPTION
**Pull request by cfchase**

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -93,35 +88,48 @@ In this tutorial, we'll create an APB for a containerized [hello world applicati
...
 description: "My New APB"
-bindable: false
+bindable: true
 async: optional
+parameters: []
 ```
 ```dockerfile
 # Dockerfile
 FROM ansibleplaybookbundle/apb-base
+# MAINTAINER {{ $MAINTAINER }}
```
Comment (rthallisey) [inline docs/getting_started.md]: Let's set this to ```MAINTAINER ansibleplaybookbundle community```

File: docs/getting_started.md [hunk 2]
```diff
@@ -156,25 +164,26 @@ apb build <docker-org>/my-apb
...
 ├── apb.yml
 ├── Dockerfile
 ├── playbooks
-│   └── provision.yml
+│   └── provision.yml # edit this file
 └── roles
 ```
 `playbooks/provision.yml` is the Ansible playbook that will be run when the **_provision_** action is called from the Ansible Service Broker.  Paste in the following code:
 ```yaml
+# playbooks/provision.yml
```
Comment (rthallisey) [inline docs/getting_started.md]: do we need this comment?

File: docs/getting_started.md [hunk 3]
```diff
@@ -430,6 +463,259 @@ In `provision.yml`, created earlier, we created the deployment config, service,
...
+    - name: port-5432
+      port: 5432
+      protocol: TCP
+      target_port: 5432
+  register: postgres_service      
+```
+Edit the `roles/deprovision-my-pg-apb/tasks/main.yml` and delete the created resources.
+
+[TODO]: # (Find out how to delete the persistent volume.  Find out about leftover replication controllers)
+```yaml
```
Comment (rthallisey) [inline docs/getting_started.md]: ```oc delete pv <my_pv>```

File: docs/getting_started.md [hunk 4]
```diff
@@ -430,6 +463,259 @@ In `provision.yml`, created earlier, we created the deployment config, service,
...
+##### Bind - Encode credentials
+[TODO]: # (Revisit these instructions in the case we hide the implemetation of the bind better)
+To make the Postgres credentials available to other applications, we need to encode them and then store them so that they'll be available.  Edit the `roles/provision-my-pg-apb/tasks/main.yml` and add the following to the very end:
+```yaml
+- name: encode bind credentials
+  shell: 'echo "{\"POSTGRES_HOST\": \"{{ postgres_service.service.spec.cluster_ip }}\", \"POSTGRES_PORT\": \"5432\", \"POSTGRES_USER\": \"{{ postgresql_user }}\", \"POSTGRES_PASSWORD\": \"{{ postgresql_password }}\", \"POSTGRES_DB\": \"{{ postgresql_database }}\"}" | base64 -w 0'
+  register: encoded_bind_credentials
+
+- debug:
+    msg: "<BIND_CREDENTIALS>{{ encoded_bind_credentials.stdout }}</BIND_CREDENTIALS>"```
```
Comment (rthallisey) [inline docs/getting_started.md]: The broker bind output is changing per https://github.com/fusor/apb-examples/pull/25.  So this code will change to https://github.com/fusor/apb-examples/pull/25/commits/f56c0a6bdad4c87fb3a89247c138b874f3974de8

**Review by rthallisey — CHANGES_REQUESTED** [review]
👍 nice doc

File: docs/getting_started.md (hunk 1 above)
Comment (cfchase) [reply docs/getting_started.md]: This is just a paste showing what `apb init` is generating.  I'd rather not change it, and it was just extra info if I make the user change it.

**Comment by cfchase** [comment]
@rthallisey I think I've addressed most of your comments.  I can remove the `# MAINTAINER` comment in the Dockerfile if you think that makes more sense.

**Comment by djwhatle** [comment]
Just attempted running catasb using these instructions and found the following undocumented steps for setup on Fedora 25.

- Create `/etc/docker/daemon.json` to specify the IP of my DNS server, since Google DNS (8.8.8.8, the Docker default) is blocked on my network.


```
$ sudo vi  /etc/docker/daemon.json
```
```
{                                                                               
    "dns": ["X.X.X.X"]                                                       
}  
```

- Give user account sudo-less access to the Docker socket.

```
$ sudo groupadd docker
$ sudo chown root:docker /var/run/docker.sock
$ sudo usermod -aG docker dwhatley
```

 - Add the OpenShift registry as insecure registry to `/etc/sysconfig/docker`, restart docker daemon.
```
$ sudo vi  /etc/sysconfig/docker
```
```
...  
# INSECURE_REGISTRY='--insecure-registry'                                       
INSECURE_REGISTRY='--insecure-registry 172.30.0.0/16'  
...
```

```
$ sudo systemctl restart docker
```

As of right now I still haven't been able to bring up the catalog and broker successfully. I'll write a new comment if I uncover something new.

**Comment by rthallisey** [comment]
LGTM.  I think the maintainer comment can be fixed from the apb tool side then we can come back and change it.

**Comment by jmrodri** [comment]
@djwhatle I didn't have to do anything with daemon.json on either of my machines.

I did have to add myself to the docker group, but didn't have to chmod any socket. I just did a `dnf install docker`

Then I followed the OC instructions to use `oc cluster up`.

https://github.com/openshift/origin/blob/master/docs/cluster_up_down.md#linux

**Comment by jmrodri** [comment]
I would maybe add a prerequisite that you have to have docker and oc installed and setup. Then maybe a troubleshooting section somewhere where it lists out things we did to solve common problems. So adding daemon.json if you can't access dns, etc. But I think it might be different for different machine and could clutter this document which is focused on apbs.

**Comment by cfchase** [comment]
@jmrodri @djwhatle added a note about installing OpenShift Origin. https://github.com/fusor/ansible-playbook-bundle/pull/39/commits/a356ae3b86739966356e6fe4e3676b6151e1ba88

@rthallisey sounds good, I'll merge after you clear the change request.  I took the maintainer line out in https://github.com/fusor/ansible-playbook-bundle/pull/40 and we can replace it with something that makes more sense when we decide what that is.  Maybe using the --org value since the user has to enter that.

**Pull request merged**